lib/AArch64Base.ml

```diff
@@ -1676,9 +1686,23 @@ let do_pp_instruction m =
   | I_SWPBH (bh,rmw,r1,r2,r3) ->
       sprintf "%s %s,%s,[%s]" (swpbh_memo bh rmw) (pp_wreg r1) (pp_wreg r2) (pp_xreg r3)
 (* Fecth and Op *)
+  | I_LDOP (op,v,(RMW_P|RMW_L as rmw),r1,ZR,r2) ->
+     if m.compat then
+       sprintf "%s %s,%s,[%s]"
+        (ldop_memo op rmw) (pp_vreg v r1) (pp_vreg v ZR) (pp_xreg r2)
+     else
+       sprintf "%s %s,[%s]"
+         (stop_memo op (rmw_to_w rmw)) (pp_vreg v r1) (pp_xreg r2)
```

Comment:
Wouldn't this print a test with LDADD Xs,XZR,[Xn] print it as an STADD (for example in the litmus7 output), do we want that?

Reply:
I guess we do, as documentation tells that aliases are "always the preferred disassembly".

Reply:
oh I see, sorry I missed it. Very well spotted!